### PR TITLE
instruct preset and config to be aligned with llama3 models 

### PR DESCRIPTION
**Pull request by uralik**
**What does this PR do? Please describe:**
* llama3 to be the default preset
* align default instruction config with llama3 model and preset name
* added grad accumulation 2 by default 

**Does your PR introduce any breaking changes? If yes, please list them:**

No 

**Check list:**
- [ ] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**?
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

File: src/fairseq2/recipes/lm/instruction_finetune.py [hunk 1]
```diff
@@ -54,16 +54,16 @@ class InstructionFinetuneConfig:
     """Holds the configuration of an instruction-finetuning recipe."""
 
     # Data
-    dataset_name: str = "openeft"  # TODO: fix!
+    dataset_name: str = "generic_instruction"
```
Comment (cbalioglu) [inline src/fairseq2/recipes/lm/instruction_finetune.py:57]: `generic_instruction` is the family name. This should point to a dataset name though. Until we have some open source dataset, we can keep "openeft" as placeholder.

File: src/fairseq2/recipes/lm/instruction_finetune.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ class InstructionFinetuneConfig:
     num_lr_warmup_steps: int = 1  # TODO: 2000
     """The number of learning rate warm-up steps."""
 
-    gradient_accumulation: int = 1
+    gradient_accumulation: int = 2
```
Comment (cbalioglu) [inline src/fairseq2/recipes/lm/instruction_finetune.py:116]: Is this intentional?

**Review by cbalioglu — COMMENTED** [review]
Thanks a lot for the PR! Left to small comments.

File: src/fairseq2/recipes/lm/instruction_finetune.py (hunk 2 above)
Comment (uralik) [reply src/fairseq2/recipes/lm/instruction_finetune.py:116]: yeah I figured that its always good to have that batch order randomness in by default when we have fixed content pre-bucketed batches
no strong opinion though

File: src/fairseq2/recipes/lm/instruction_finetune.py (hunk 1 above)
Comment (uralik) [reply src/fairseq2/recipes/lm/instruction_finetune.py:57]: ah true! will revert

**Review by cbalioglu — APPROVED** [review]
LGTM!

**Pull request merged**